standalone/inc/pup/SDL_ttf.c

```diff
@@ -0,0 +1,4223 @@
+/*
```

Comment:
Any reason to not pull this in as a dependency?

Reply:
It's a single .c file, the build is a bit complex it looked simpler to just copy it.

Reply:
SDL_ttf used to be part of the build. But now I see it was as binary so also not ideal.

54d8b8400b5763f0e64d3edb3084c9c0be65eed9
224c661129b36cc90e218c42f63da22016319b15

Reply:
SDL_ttf wasn't being used by anything, but we removed it. Since this will use it, that code will come back. We do want everything as libs and so's.

Reply:
> SDL_ttf wasn't being used by anything, but we removed it. Since this will use it, that code will come back. We do want everything as `.dylib` and `.so`.

Not sure what you want to say here? There are 3 options:

* copy over code to the repo like this pr does.
* store binaries of libraries in the repo like used to be the case for SDL_ttf (so/dll/dylib)
* fetching dependencies as binary or  fetch code and compile like external.sh does

Which would be preferred here? I would think the third option is preferable?

Reply:
Sorry for not being clear enough. 

we will update this PR and build upon it to do the `external.sh` way.

Reply:
I will look into it this weekend. After a bit of Googling the only reference about the problem I'm seeing on Linux is for someone else using SDL_ttf. I will try to build it without SDL_ttf and test it on Linux. If I don't see the issue anymore I might remove it all together and call FreeType directly, otherwise I will make it a dependency.

@jsm174, any thoughts on using cmake for dependency management instead of a shell file? I played a bit with FetchContent_Declare to download a zip/tar and build it. Using OVERRIDE_FIND_PACKAGE makes sure cmake doesn't grab a version that might be installed in the system.

Reply:
Yes, I thought about using cmake to fetch dependencies. But the differences between building for different platforms, (for example compiling libserialport), it just makes the cmake files too complicated, and they are already too complicated.

@vbousquet started moving things around into `third-party/runtime-libs` and `third-party/build-libs` and I think this is exactly what we needed.

If you look at the build processes for the following projects:

https://github.com/vpinball/libaltsound
https://github.com/jsm174/libdmdutil
https://github.com/ppuc/libzedmd/tree/libserialport

We tried to stick to the same approach, and it seems to work well.

I'm going to start slowly moving standalone to this as well.

Reply:
I updated the PR and moved SDL_ttf to be built in external.sh along with other dependencies